coalib/misc/DictUtilities.py

```diff
@@ -32,10 +32,7 @@ def add_pair_to_dict(key, value, dictionary):
     Add (key, value) pair to the dictionary. The value is added to a list of
     values for the key.
     """
-    if key in dictionary:
-        dictionary[key].append(value)
-    else:
-        dictionary[key] = [value]
+    dictionary[key].append(value)
```

Comment:
This is a public function, which means it can be used by other code which you cant see.
Your change requires that the parameter `dictionary` is a `defaultdict(list)` , which is not a guarantee.

You should use `dictionary[key].append(value)` in `inverse_dicts` , and deprecate `add_pair_to_dict`.